### PR TITLE
Bug 1286939 - Instructions for registering a Pulse Job Exchange

### PR DESCRIPTION
**Pull request by camd**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1690)

### CODE REVIEW

File: docs/submitting_pulse_data.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+With `Pulse Guardian`_, you can create your Pulse User in order to access and
+create your own Queues and Exchanges.  There is no mechanism to create an
+Exchange in the Pulse Guardian UI itself, however.  You will need to create
+your exchange in your submitting code.  There are a few options available
+for that:
+
+1. `MozillaPulse`_
+2. `Kombu`_
+3. `librabbitmq`_
```
Comment (edmorley) [inline docs/submitting_pulse_data.rst]: librabbitmq is not very well maintained, let's drop it from the list :-)

File: docs/index.rst [hunk 2]
```diff
@@ -18,6 +18,7 @@ Contents:
    list_of_services
    common_tasks
    rest_api
+   submitting_pulse_data
```
Comment (edmorley) [inline docs/index.rst]: How about a name like 'Submitting via Pulse' or similar?

"Submitting Pulse data" make me think of submitting data, where the data is describing pulse, which is something different :-)

Comment (wlach) [inline docs/index.rst]: Could we just make them two subsections under "Submitting Data": 

* Submitting using the Python client
* Submitting using Pulse

Then you could have an intro section under "submitting data" describing the advantages/disadvantages of each.

IMO it's very confusing to put this information in two different sections. Someone looking at the Treeherder documentation for the first time will very likely have no idea what "pulse" is.

Comment (edmorley) [inline docs/index.rst]: Agree that would be better :-)

Comment (camd) [inline docs/index.rst]: Yep, good call.  I reorganized things a bit and added a brief synopsis to the ref links for pulse vs. API.

File: docs/submitting_data.rst [hunk 3]
```diff
@@ -1,9 +1,122 @@
...
+
+To submit via a Pulse exchange, these are the steps you will need to follow:
+
+1. Format your data
+-------------------
+
+The new job format in `YML Schema`_ is significantly different from the one
+used in our API job submission.  It has virtually the same data, just in
+(what we hope) is a better form.  You are responsible for validating your data
+prior to publishing it onto your Exchange, or Treeherder may reject it.
```
Comment (wlach) [inline docs/submitting_data.rst]: I would not bother contrasting the pulse data with the API data here. Also I would say `Pulse Job Schema` as opposed to `YML Schema`.

Maybe:

```
You should format your job data according to the `Pulse Job Schema`_, which describes the various properties of a job: whether it passed or failed, job group/type symbol, description, log information, etc. You are responsible for validating your data prior to publishing it onto your exchange, or Treeherder may reject it.
```

**Comment by wlach** [comment]
Thanks Cam! I think this organization is a lot better, still one comment. :)

**Pull request merged**